pkg/controller/atlasproject/team_reconciler.go

```diff
@@ -295,3 +327,37 @@ func renameTeam(ctx context.Context, workflowCtx *workflow.Context, atlasTeam *m
 
 	return atlasTeam, nil
 }
+
+func teamsManagedByAtlas(ctx context.Context, atlasClient mongodbatlas.Client, orgID string) customresource.AtlasChecker {
+	return func(resource v1.AtlasCustomResource) (bool, error) {
+		team, ok := resource.(*v1.AtlasTeam)
+		if !ok {
+			return false, errors.New("failed to match resource type as AtlasTeams")
+		}
+
+		if team.Status.ID == "" {
+			return false, nil
+		}
```

Comment:
Not sure about this. I mean, the status can be unset in k8s and it does only mean that it has not synced with Atlas. Or maybe I am missing something.